src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/database/JdbcQuestionRepository.java

```diff
@@ -34,21 +34,30 @@ private DataSource getDataSource() {
     @Override
     public Collection<Question> findBy(QuestionQuery query) {
         setup(dataSource);
-        if (query.getShouldContain() != null) {
-            return findAll().stream()
-                    .filter(question -> (
-                            question.getTitle() != null && question.getTitle()
-                                    .toLowerCase()
-                                    .contains(query.getShouldContain().toLowerCase())
-                    ) || (
-                            question.getDescription() != null && question.getDescription()
-                                    .toLowerCase()
-                                    .contains(query.getShouldContain().toLowerCase())
-                    ))
-                    .collect(Collectors.toList());
-        } else {
-            return findAll();
+        var select = "SELECT * FROM Question WHERE title = ? OR description = ?;";
```

Comment:
We are not looking for exact matches, but rather contents that contain the query with arbitrary case.